.github/workflows/ci.yml

```diff
@@ -162,3 +162,48 @@ jobs:
           pip install tox
           tox -c examples/linalg
           tox -c examples/parallel
+
+  coverage:
+    name: coverage-${{ matrix.platform.os }}-${{ matrix.platform.arch }}-py${{ matrix.python }}
+    strategy:
+      matrix:
+        python: [3.9]
+        platform: [
+          { os: ubuntu, arch: x64 },
+        ]
+    runs-on: ${{ matrix.platform.os }}-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+          architecture: ${{ matrix.platform.arch }}
+      - name: Install numpy
+        run: pip install numpy
+      - uses: Swatinem/rust-cache@v1
+        with:
+          key: coverage-cargo-${{ matrix.platform.os }}
+        continue-on-error: true
+      - uses: actions-rs/toolchain@v1
+        with:
+          # TODO: this is pinned to work around a rustc bug in instrumented coverage
+          # https://github.com/taiki-e/cargo-llvm-cov/issues/128
+          toolchain: nightly-2022-01-14
+          override: true
+          profile: minimal
+          components: llvm-tools-preview
+      - name: install cargo-llvm-cov
+        shell: bash
+        run: |
+          host=$(rustc -Vv | grep host | sed 's/host: //')
+          curl -fsSL https://github.com/taiki-e/cargo-llvm-cov/releases/download/v${CARGO_LLVM_COV_VERSION}/cargo-llvm-cov-"$host".tar.gz | tar xzf - -C ~/.cargo/bin
+        env:
+          CARGO_LLVM_COV_VERSION: 0.1.16
+      - run: |
+          cargo llvm-cov clean
+          cargo llvm-cov --output-path coverage.lcov
```

Comment:
I think you might need `--lcov` flag here too to produce an output in the correct format? I'm not sure what gets generated without it.

Codecov appears to have an error, so I suspect adding this flag might resolve?